build_conan.py

```diff
@@ -0,0 +1,81 @@
+import os
+import sys
+import subprocess
+
+
+if __name__ == '__main__' :
+
+  # Defaulted env var inputs - can override if necessary
+  CONAN = os.environ.get('CONAN', 'conan')
+  CONAN_HOME = os.environ.get('CONAN_HOME', f'{os.getcwd()}/.conan2')
+  os.environ['CONAN_HOME'] = CONAN_HOME # export it
+  OUTPUT_DIR = os.environ.get('OUTPUT_DIR', f'{os.getcwd()}/build')
+  CONAN_BUILD_PROFILE = os.environ.get('CONAN_BUILD_PROFILE', 'default')
+  CONAN_HOST_PROFILE = os.environ.get('CONAN_HOST_PROFILE', 'default')
+  SLM_BUILD_DYNAMIC_LIB = os.environ.get('SLM_BUILD_DYNAMIC_LIB', 'False').capitalize()
+
+  def no_conan_profile () :
+    return (CONAN_BUILD_PROFILE == 'default' or CONAN_HOST_PROFILE == 'default') and not (os.path.exists(os.path.join(CONAN_HOME, 'profiles', 'default')))
+
+  if sys.prefix == sys.base_prefix :
+    print('ERROR: This script is intended to run in a pre-existing python virtual environment')
+    print('       Please activate a python venv and run this script again')
+    sys.exit(-1)
+
+  print(f'                CONAN: {CONAN}')
+  print(f'           CONAN_HOME: {CONAN_HOME}')
+  print(f'           OUTPUT_DIR: {OUTPUT_DIR}')
+  print(f'  CONAN_BUILD_PROFILE: {CONAN_BUILD_PROFILE}')
+  print(f'   CONAN_HOST_PROFILE: {CONAN_HOST_PROFILE}')
+  print(f'SLM_BUILD_DYNAMIC_LIB: {SLM_BUILD_DYNAMIC_LIB}')
+  print(f"          VIRTUAL_ENV: {os.environ.get('VIRTUAL_ENV')}")
+
+  # TODO: is there something smarter to do?
+  subprocess.run([sys.executable, '-m', 'pip', 'install', '-r' 'requirements.txt'],
```

Comment:
At least for this repository - this is unnecessary. The `requirements.txt` file is really only there to support installing `conan` if it doesn't exist. I think this script is making the assumption that `conan` is present in the environment - correct?

The `wrapper_requirements.txt` is not required to build the dependency - it is only needed to generate the `Wrapper.stanza` file from the Z3 header file.

Reply:
If we move the `LBStanzaGenerator` and `lbstanza_deployer` to a pip installable package then this `pip install` becomes pertinent again.